src/main/scala/scalaz/stream/wye.scala

```diff
@@ -836,9 +836,7 @@ object wye {
         }
       })(S)
 
-      repeatEval(Task.async[Seq[O]] { cb => a ! Get(cb) })
-      .flatMap(emitAll)
-      .onComplete(eval_(Task.async[Unit](cb => a ! DownDone(cb))))
+      repeatEval(Task.async[Seq[O]] { cb => a ! Get(cb) }) onHalt { _.asHalt } flatMap emitAll onComplete eval_(Task.async[Unit](cb => a ! DownDone(cb)))
```

Comment:
Having long lines is bad.  No argument there.  The only real downside of infix notation is that it requires some inversion of arguments (noun-verb) in order to break lines, which is deeply unfortunate.

I don't have too much of a problem choosing one consistent code style, though I make absolutely no secret that I have a very hard time reading certain Scala styles, and I don't doubt that writing them would be even more difficult.  The existing code style within scalaz-stream, even in places where a code block is written by a single author is inconsistent to say the least.  I'm very much in favor of being consistent about things, but my impression was that consistency in the style wasn't much cared for.  (that's not a slight, some developers just don't care about stylistic consistency, and that's fine)  It was for that reason that I made no attempt to retain any consistency with existing code.

> Just for curiosity whats wrong with writing your code as below?

All of those lines are indented equally, making it difficult to read them as part of the same invocation.  The spacing on the parentheses is inconsistent even within the same construct (see: the lambdas on the first and last lines).  The use of parentheses for lambdas *necessitates* inconsistency with lambdas that have variables or statements in them, and thus as a general rule, *no* internally-consistent code style allows parenthetical lambdas.  The corollary to the parenthetical lambda issue derives from the inconsistent association of dots relative to curly braces vs parentheses (using a dot dispatch on a curly brace block often, though not always, associates to the result of the block rather than to the result of the invocation taking the block as a parameter).  Thus, using braces for blocks (which again is *required* for an internally-consistent style) categorically rules out using dot dispatch ubiquitously, since dots cannot be consistently used in the presence of curly braces.